### DISCUSSION TITLE
share my answer for "2.5. Exercises for TensorIR"

### DISCUSSION THREAD

**Discussion by hbsun2113** [mlc-ai/mlc-en#17] [General]
If anything is wrong, please let me know, thanks!
https://colab.research.google.com/gist/hbsun2113/ff3a7d96a1f83904f1d9aeab61b9e836/2-5-exercises-for-tensorir-ipynb.ipynb

Also attach the [exercises](https://mlc.ai/chapter_tensor_program/tensorir_exercises.html)

**Comment by Zhaoyilunnn** [comment]
Great job! 
BTW why should we parrallel n first, I got errors like this if I do not parallel n first
```
Error message: The queried subtree root tir.For#0 in SRef tree does not have compact dataflow, because its child block tir.Block#1 on SRef tree is neither a local complete block nor a local reduction block.
```

**Comment by meicale** [comment]
> If anything is wrong, please let me know, thanks! https://colab.research.google.com/gist/hbsun2113/ff3a7d96a1f83904f1d9aeab61b9e836/2-5-exercises-for-tensorir-ipynb.ipynb
> 
> Also attach the [exercises](https://mlc.ai/chapter_tensor_program/tensorir_exercises.html)

Have you ever tried " A: T.Buffer[(N, CI, H, W), "int64"]" ? And this gives me an error. Can you figure it out?

**Comment by kevinliupy** [comment]
Great job!
